### PR TITLE
Key Vault: switched all instances of vault_endpoint to vault_url

### LINKED ISSUE #8120: Rename vaultEndpoint to vaultUri/vaultUrl (idiomatic)
After discussions with the arch board, we have decided to rename `vaultEndpoint` (parameters, properties, etc.) to `vaultUri` or `vaultUrl` (idiomatic). This is true for all clients and models where this exists across:

- [ ] Secrets
- [ ] Keys
- [ ] Certificates

### PR DESCRIPTION
**Pull request by iscai-msft**
fixes #8120

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/models.py [hunk 1]
```diff
@@ -230,13 +230,13 @@ def recovery_level(self):
         return self._attributes.recovery_level if self._attributes else None
 
     @property
-    def vault_endpoint(self):
+    def vault_url(self):
         # type: () -> str
         """The name of the vault that the certificate is created in.
```
Comment (chlowell) [inline sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/models.py]: Needn't be fixed in this PR: ~name~ URL

File: sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/models.py [hunk 2]
```diff
@@ -1130,13 +1130,13 @@ def provider(self):
         return self._provider
 
     @property
-    def vault_endpoint(self):
+    def vault_url(self):
         # type: () -> str
         """The name of the vault with this issuer.
```
Comment (chlowell) [inline sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/models.py]: (also here, now I can't un-see it!)

File: sdk/keyvault/azure-keyvault-keys/HISTORY.md [hunk 3]
```diff
@@ -38,8 +38,8 @@ as certain properties of `Key` (such as `version`) have to be accessed
 through the `properties` property. See the updated [docs](https://azure.github.io/azure-sdk-for-python/ref/azure.keyvault.keys.html)
 for details.
 - `update_key` has been renamed to `update_key_properties`
-- The `vault_url` parameter of `KeyClient` has been renamed to `vault_endpoint`
-- The property `vault_url` has been renamed to `vault_endpoint` in all models
+- The `vault_url` parameter of `KeyClient` has been renamed to `vault_url`
```
Comment (chlowell) [inline sdk/keyvault/azure-keyvault-keys/HISTORY.md]: :suspect:

File: sdk/keyvault/azure-keyvault-secrets/HISTORY.md [hunk 4]
```diff
@@ -25,8 +25,8 @@ as certain properties of `Secret` (such as `version`) have to be accessed
 through the `properties` property. See the updated [docs](https://azure.github.io/azure-sdk-for-python/ref/azure.keyvault.secrets.html)
 for details.
 - `update_secret` has been renamed to `update_secret_properties`
-- The `vault_url` parameter of `SecretClient` has been renamed to `vault_endpoint`
-- The property `vault_url` has been renamed to `vault_endpoint` in all models
+- The `vault_url` parameter of `SecretClient` has been renamed to `vault_url`
+- The property `vault_url` has been renamed to `vault_url` in all models
```
Comment (chlowell) [inline sdk/keyvault/azure-keyvault-secrets/HISTORY.md]: :suspect:

File: sdk/keyvault/azure-keyvault-certificates/HISTORY.md [hunk 5]
```diff
@@ -32,8 +32,8 @@ as certain properties of `Certificate` (such as `version`) have to be accessed
 through the `properties` property. See the updated [docs](https://azure.github.io/azure-sdk-for-python/ref/azure.keyvault.certificates.html)
 for details.
 - `update_certificate` has been renamed to `update_certificate_properties`
-- The `vault_url` parameter of `CertificateClient` has been renamed to `vault_endpoint`
-- The property `vault_url` has been renamed to `vault_endpoint` in all models
+- The `vault_url` parameter of `CertificateClient` has been renamed to `vault_url`
+- The property `vault_url` has been renamed to `vault_url` in all models
```
Comment (chlowell) [inline sdk/keyvault/azure-keyvault-certificates/HISTORY.md]: :suspect:

File: sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/models.py (hunk 1 above)
Comment (iscai-msft) [reply sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/models.py]: fixed, thanks!

File: sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/models.py (hunk 2 above)
Comment (iscai-msft) [reply sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/models.py]: fixed, thanks!

File: sdk/keyvault/azure-keyvault-secrets/HISTORY.md (hunk 4 above)
Comment (iscai-msft) [reply sdk/keyvault/azure-keyvault-secrets/HISTORY.md]: oh god that was bad, fixed now

File: sdk/keyvault/azure-keyvault-keys/HISTORY.md (hunk 3 above)
Comment (iscai-msft) [reply sdk/keyvault/azure-keyvault-keys/HISTORY.md]: oh god this is bad, thanks

File: sdk/keyvault/azure-keyvault-certificates/HISTORY.md (hunk 5 above)
Comment (iscai-msft) [reply sdk/keyvault/azure-keyvault-certificates/HISTORY.md]: oh no, sorry :'(

**Pull request merged**